src/posix/platform/infra_if.cpp

```diff
@@ -250,6 +250,9 @@ void platformInfraIfInit(otInstance *aInstance, const char *aIfName)
         DieNow(OT_EXIT_ERROR_ERRNO);
     }
 
+    rval = setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE, sInfraIfName, strlen(sInfraIfName));
```

Comment:
also handle the case `__NetBSD__ || __FreeBSD__ || __APPLE__` ?

Reply:
Current infra IF implementation generally supports only linux. I think it is better to leave it in a future to add support for BSD and APPLE platforms.